lib/rgeo/shapefile/reader.rb

```diff
@@ -342,7 +343,10 @@ def next
       # and yield the Reader::Record for each one.
 
       def each
+        return to_enum(:each) unless block_given?
```

Comment:
An enumerator must have a definition of it's on size, otherwise, `enumurator.size == nil`. I think here it would be pretty trivial:

```suggestion
        return to_enum(:each) { @num_records } unless block_given?
```